app/controllers/dois_controller.rb

```diff
@@ -31,6 +31,16 @@ def index
 
     page = page_from_params(params)
 
+    # For DOI's we need to decode the page cursor
+    if page[:cursor].present?
```

Comment:
I would implement this here: https://github.com/datacite/lupo/blob/master/app/controllers/concerns/paginatable.rb#L6-L29

Reply:
I didn't do this because paginatable is something that applies across everything that implements it. I kept it as being something that just returns the cursor from the paras into the "page", it doesn't therefore know it's a token or anything, that's up to individual controllers to decide if it it needs to do additional work. This is also why I force a sort here as this is specific to dois handling.

Reply:
Don't we want to support cursor pagination everywhere? Or do you think it is really only needed for the large indexes `dois` and `events`? In the latter case I agree with your approach.